bool can_send
---
bool success
